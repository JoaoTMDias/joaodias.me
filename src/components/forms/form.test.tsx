import React from "react";
import user from "@testing-library/user-event";
import { render, cleanup, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Form from "./form";

afterEach(cleanup);

describe("<Form />", () => {
	it("should render without errors", () => {
		const component = render(<Form />);
		expect(component).toMatchSnapshot();
	});

	it("should have the submit button disabled by default", async () => {
		const { findByTestId } = render(<Form />);
		const button = await findByTestId("submit-button");

		expect(button).toHaveAttribute("aria-disabled", "true");
	});

	it("should enable the submit button once the form is filled", async () => {
		const { getByTestId } = render(<Form />);
		const name = await getByTestId("component-text-input-name");
		const email = await getByTestId("component-text-input-email");
		const message = await getByTestId("component-text-area-input");
		const button = await getByTestId("submit-button");

		user.type(name, "name input");
		user.type(email, "email@email.com");
		user.type(message, "textarea message content here");

		await waitFor(() => {
			expect(button).toHaveAttribute("aria-disabled", "false");
		});
	});
});
