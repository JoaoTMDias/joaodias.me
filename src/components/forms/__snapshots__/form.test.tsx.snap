// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="layout__row"
        id="contact-form"
      >
        <form
          action="/contact"
          aria-label="Contact form. Includes a name,email a message inputs."
          class="sc-AxiKw doLXDD"
          data-netlify="true"
          data-netlify-honeypot="bot-field"
          data-testid="contact-page-form"
          method="post"
          name="contact-page-form"
        >
          <input
            name="form-name"
            type="hidden"
            value="contact-page-form"
          />
          <div
            aria-hidden="false"
            class="sc-AxhUy bETLpA"
            role="group"
          >
            <div
              class="sc-AxjAm cktUFz"
              data-form="input"
              data-testid="component-text-wrapper"
            >
              <label
                aria-labelledby="Name-name"
                class="content"
                data-testid="component-text-label"
                for="name"
              >
                <span
                  class="label"
                  id="Name-name"
                >
                  Name
                </span>
                <input
                  class="input undefined"
                  data-testid="component-text-input-name"
                  id="name"
                  maxlength="50"
                  minlength="3"
                  name="name"
                  placeholder="What's your name?"
                  required=""
                  type="text"
                  value=""
                />
                
              </label>
            </div>
            <div
              class="sc-AxjAm cktUFz"
              data-form="input"
              data-testid="component-text-wrapper"
            >
              <label
                aria-labelledby="Email-email"
                class="content"
                data-testid="component-text-label"
                for="email"
              >
                <span
                  class="label"
                  id="Email-email"
                >
                  Email
                </span>
                <input
                  class="input undefined"
                  data-testid="component-text-input-email"
                  id="email"
                  maxlength="50"
                  minlength="5"
                  name="email"
                  placeholder="Your fancy email"
                  required=""
                  type="email"
                  value=""
                />
                
              </label>
            </div>
            <div
              class="sc-AxjAm sc-AxirZ kOZIao"
              data-form="textarea"
              data-testid="component-text-area-wrapper"
            >
              <label
                class="content"
                data-testid="component-text-area-label"
                for="message"
              >
                <span
                  class="label"
                >
                  Message
                </span>
                <textarea
                  class="input undefined"
                  cols="500"
                  data-testid="component-text-area-input"
                  id="message"
                  maxlength="500"
                  minlength="10"
                  name="message"
                  placeholder="Got something you want to say?"
                  required=""
                />
                
              </label>
            </div>
            <input
              class="hidden"
              hidden=""
              name="bot-field"
            />
          </div>
          <div
            aria-hidden="false"
            class="sc-AxhUy bETLpA"
          >
            <button
              aria-disabled="true"
              class="sc-AxhCb fslhpO"
              data-testid="submit-button"
              id="submit"
              type="submit"
            >
              Send Message
            </button>
          </div>
          <div
            data-netlify-recaptcha="true"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="layout__row"
      id="contact-form"
    >
      <form
        action="/contact"
        aria-label="Contact form. Includes a name,email a message inputs."
        class="sc-AxiKw doLXDD"
        data-netlify="true"
        data-netlify-honeypot="bot-field"
        data-testid="contact-page-form"
        method="post"
        name="contact-page-form"
      >
        <input
          name="form-name"
          type="hidden"
          value="contact-page-form"
        />
        <div
          aria-hidden="false"
          class="sc-AxhUy bETLpA"
          role="group"
        >
          <div
            class="sc-AxjAm cktUFz"
            data-form="input"
            data-testid="component-text-wrapper"
          >
            <label
              aria-labelledby="Name-name"
              class="content"
              data-testid="component-text-label"
              for="name"
            >
              <span
                class="label"
                id="Name-name"
              >
                Name
              </span>
              <input
                class="input undefined"
                data-testid="component-text-input-name"
                id="name"
                maxlength="50"
                minlength="3"
                name="name"
                placeholder="What's your name?"
                required=""
                type="text"
                value=""
              />
              
            </label>
          </div>
          <div
            class="sc-AxjAm cktUFz"
            data-form="input"
            data-testid="component-text-wrapper"
          >
            <label
              aria-labelledby="Email-email"
              class="content"
              data-testid="component-text-label"
              for="email"
            >
              <span
                class="label"
                id="Email-email"
              >
                Email
              </span>
              <input
                class="input undefined"
                data-testid="component-text-input-email"
                id="email"
                maxlength="50"
                minlength="5"
                name="email"
                placeholder="Your fancy email"
                required=""
                type="email"
                value=""
              />
              
            </label>
          </div>
          <div
            class="sc-AxjAm sc-AxirZ kOZIao"
            data-form="textarea"
            data-testid="component-text-area-wrapper"
          >
            <label
              class="content"
              data-testid="component-text-area-label"
              for="message"
            >
              <span
                class="label"
              >
                Message
              </span>
              <textarea
                class="input undefined"
                cols="500"
                data-testid="component-text-area-input"
                id="message"
                maxlength="500"
                minlength="10"
                name="message"
                placeholder="Got something you want to say?"
                required=""
              />
              
            </label>
          </div>
          <input
            class="hidden"
            hidden=""
            name="bot-field"
          />
        </div>
        <div
          aria-hidden="false"
          class="sc-AxhUy bETLpA"
        >
          <button
            aria-disabled="true"
            class="sc-AxhCb fslhpO"
            data-testid="submit-button"
            id="submit"
            type="submit"
          >
            Send Message
          </button>
        </div>
        <div
          data-netlify-recaptcha="true"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
